lib/experimental/register-webfonts-from-theme-json.php

```diff
@@ -122,28 +123,28 @@ function gutenberg_add_registered_webfonts_to_theme_json( $data ) {
 		$data['settings']['typography']['fontFamilies'] = array();
 	}
 
-	// Add missing fonts.
-	foreach ( $to_add as $family ) {
-		$font_face = array();
-		foreach ( $font_families_registered as $font_family ) {
-			if ( $family !== $font_family['font-family'] ) {
-				continue;
-			}
-			$camel_cased = array();
-			foreach ( $font_family as $key => $value ) {
+	foreach ( $to_add as $slug ) {
+		$font_faces_for_family = $font_families_registered[ $slug ];
+		$family_name           = $font_faces_for_family[0]['font-family'];
+		$font_faces            = array();
+
+		foreach ( $font_faces_for_family as $font_face ) {
+			$camel_cased = array( 'origin' => 'gutenberg_wp_webfonts_api' );
+			foreach ( $font_face as $key => $value ) {
 				$camel_cased[ lcfirst( str_replace( '-', '', ucwords( $key, '-' ) ) ) ] = $value;
 			}
-			$font_face[] = $camel_cased;
+			$font_faces[] = $camel_cased;
 		}
+
 		$data['settings']['typography']['fontFamilies'][] = array(
-			'fontFamily' => false !== strpos( $family, ' ' ) ? "'{$family}'" : $family,
-			'name'       => $family,
-			'slug'       => sanitize_title( $family ),
-			'fontFace'   => $font_face,
+			'fontFamily' => false !== strpos( $family_name, ' ' ) ? "'{$family_name}'" : $family_name,
```

Comment:
Use `str_contains()` for readability:

```suggestion
			'fontFamily' => str_contains( $family_name, ' ' ) ? "'{$family_name}'" : $family_name,
```

This yields the same result as using `strpos()` but is more readable. See it in action here https://3v4l.org/VCIg4#v8.0.17.

Reply:
Resolved in https://github.com/WordPress/gutenberg/pull/39559/commits/27e7da71a053c7e21c91d83935fbb1fd460f376b.